File: 30 Days of Code/Cpp/Day18_StackandQueues.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
    //Write your code here
    
    stack<char>st;
    queue<char>a;
    public:
    
    void pushCharacter(char ch){
        st.push(ch);
    }
    
    char popCharacter(){
        char ch = st.top();
        st.pop();
        return ch;
    }
    
    void enqueueCharacter(char ch)
    {
        a.push(ch);
    }
    
    char dequeueCharacter()
    {
        char ch = a.front();
        a.pop();
        return ch;
    }

};

int main() {
    // read the string s.
    string s;
    getline(cin, s);
    
  	// create the Solution class object p.
    Solution obj;
    
    // push/enqueue all the characters of string s to stack.
    for (int i = 0; i < s.length(); i++) {
        obj.pushCharacter(s[i]);
        obj.enqueueCharacter(s[i]);
    }
    
    bool isPalindrome = true;
    
    // pop the top character from stack.
    // dequeue the first character from queue.
    // compare both the characters.
    for (int i = 0; i < s.length() / 2; i++) {
        if (obj.popCharacter() != obj.dequeueCharacter()) {
            isPalindrome = false;
            
            break;
        }
    }
    
    // finally print whether string s is palindrome or not.
    if (isPalindrome) {
        cout << "The word, " << s << ", is a palindrome.";
    } else {
        cout << "The word, " << s << ", is not a palindrome.";
    }
    
    return 0;
}

File: 30 Days of Code/Java/Day18_StackandQueues.java
import java.io.*;
import java.util.*;

public class Solution {
    // Write your code here.
    private Stack<Character>stack = new Stack();
    private Queue<Character>queue = new LinkedList<>();
    
     void pushCharacter(char ch)
     {
         stack.push(ch);
     }
     
     void enqueueCharacter(char ch)
     {
         queue.add(ch);
     }
     
     char  dequeueCharacter(){
         char ch = queue.peek();
         queue.remove();
         return ch;
     }
     
      char popCharacter(){
          return stack.pop();
      }
    

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String input = scan.nextLine();
        scan.close();

        // Convert input String to an array of characters:
        char[] s = input.toCharArray();

        // Create a Solution object:
        Solution p = new Solution();

        // Enqueue/Push all chars to their respective data structures:
        for (char c : s) {
            p.pushCharacter(c);
            p.enqueueCharacter(c);
        }

        // Pop/Dequeue the chars at the head of both data structures and compare them:
        boolean isPalindrome = true;
        for (int i = 0; i < s.length/2; i++) {
            if (p.popCharacter() != p.dequeueCharacter()) {
                isPalindrome = false;                
                break;
            }
        }

        //Finally, print whether string s is palindrome or not.
        System.out.println( "The word, " + input + ", is " 
                           + ( (!isPalindrome) ? "not a palindrome." : "a palindrome." ) );
    }
}